From dbdd64c66a5001269ceab9945d344df6d94e9dda Mon Sep 17 00:00:00 2001
From: Luke Shumaker <lukeshu@lukeshu.com>
Date: Sat, 20 May 2017 20:13:09 -0400
Subject: [PATCH 30/38] nspawn: Improve --help text

The `--help` text lies about what the `-U` flag does, and under-documents
the `--private-users` values. .  Fix that.
---
 src/nspawn/nspawn.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/nspawn/nspawn.c b/src/nspawn/nspawn.c
index d4001e5070..f1afa3a9ca 100644
--- a/src/nspawn/nspawn.c
+++ b/src/nspawn/nspawn.c
@@ -230,9 +230,13 @@ static void help(void) {
                "     --uuid=UUID            Set a specific machine UUID for the container\n"
                "  -S --slice=SLICE          Place the container in the specified slice\n"
                "     --property=NAME=VALUE  Set scope unit property\n"
-               "  -U --private-users=pick   Run within user namespace, autoselect UID/GID range\n"
-               "     --private-users[=UIDBASE[:NUIDS]]\n"
+               "     --private-users[=yes]  Run within user namespace, detect UID/GID range\n"
+               "     --private-users=UIDBASE[:NUIDS]\n"
                "                            Similar, but with user configured UID/GID range\n"
+               "     --private-users=pick   Similar, but autoselect an unused UID/GID range,\n"
+               "                            implies --private-users-chown\n"
+               "  -U                        Equivalent to --private-users=pick if the kernel\n"
+               "                            supports user namespaces; ignored otherwise\n"
                "     --private-users-chown  Adjust OS tree ownership to private UID/GID range\n"
                "     --private-network      Disable network in container\n"
                "     --network-interface=INTERFACE\n"
-- 
2.18.0

